DevEdu.API/DevEdu.Business/Services/TaskService.cs

```diff
@@ -9,39 +13,59 @@ public class TaskService : ITaskService
         private readonly ITaskRepository _taskRepository;
         private readonly ICourseRepository _courseRepository;
         private readonly IStudentAnswerOnTaskRepository _studentAnswerOnTaskRepository;
+        private readonly ITaskValidationHelper _taskValidationHelper;
 
-        public TaskService(ITaskRepository taskRepository, ICourseRepository courseRepository, IStudentAnswerOnTaskRepository studentAnswerOnTaskRepository)
+        public TaskService(
+            ITaskRepository taskRepository,
+            ICourseRepository courseRepository,
+            IStudentAnswerOnTaskRepository studentAnswerOnTaskRepository, 
+            ITaskValidationHelper taskValidationHelper
+        )
         {
             _taskRepository = taskRepository;
             _courseRepository = courseRepository;
             _studentAnswerOnTaskRepository = studentAnswerOnTaskRepository;
+            _taskValidationHelper = taskValidationHelper;
         }
 
-        public TaskDto GetTaskById(int id) => _taskRepository.GetTaskById(id);
-
-        public TaskDto GetTaskWithCoursesById(int id)
+        public TaskDto GetTaskById(int id)
```

Comment:
I had a "fresh look" on this code and now I have a suggestion for you:
this method must be deleted